libs/stats/odc/stats/_cli_pq_queue.py

```diff
@@ -0,0 +1,130 @@
+    """
+    Run Pixel Quality stats on tasks provided in a SQS queue
+
+    Task could be one of the 3 things
+
+    \b
+    1. Comma-separated triplet: period,x,y or 'x[+-]<int>/y[+-]<int>/period
+       2019--P1Y,+003,-004
+       2019--P1Y/3/-4          `/` is also accepted
+       x+003/y-004/2019--P1Y   is accepted as well
+    2. A zero based index
+    3. A slice following python convention <start>:<stop>[:<step]
+        ::10 -- every tenth task: 0,10,20,..
+       1::10 -- every tenth but skip first one 1, 11, 21 ..
+        :100 -- first 100 tasks
+
+    If no limit is provided all tasks in the queue will be processed.
+    E.g:
+        odc-stats run-pq-queue  s3://deafrica-stats-processing/orchestration_test/s2_l2a_2020--P1Y.db  deafrica-prod-eks-stats-geomedian \
+                  --limit 2 --output-location s3://deafrica-stats-processing/orchestration_test/output/
+    """
+
+    from tqdm.auto import tqdm
+    from functools import partial
+    from .io import S3COGSink
+    from ._pq import pq_input_data, pq_reduce, pq_product
+    from .proc import process_tasks
+    from .tasks import TaskReader
+    from datacube.utils.dask import start_local_dask
+    from datacube.utils.rio import configure_s3_access
+
+    # config
+    resampling = "nearest"
+    COG_OPTS = dict(compress="deflate", predict=2, zlevel=6, blocksize=800)
+    # ..
+
+    # TODO: Handle cache file from LOCAL or S3, maybe using
+    # https://github.com/opendatacube/datacube-alchemist/blob/main/datacube_alchemist/worker.py#L44
+    rdr = TaskReader(cache_file)
+    product = pq_product(location=output_location)
+
+    if verbose:
+        print(repr(rdr))
+
+    def get_task_from_message(message):
+        if "," in message.body:
+            msg = message.body.split(",")
+            message = (msg[0], int(msg[1]), int(msg[2]))
+            return message
+
+    def pq_proc(task):
+        ds_in = pq_input_data(task, resampling=resampling)
+        ds = pq_reduce(ds_in)
+        return ds
+
+    sink = S3COGSink(cog_opts=COG_OPTS, public=public)
+    if product.location.startswith("s3:"):
+        if not sink.verify_s3_credentials():
+            print("Failed to load S3 credentials")
+            sys.exit(2)
+
+    if verbose and sink._creds:
+        creds_rw = sink._creds
+        print(f"creds: ..{creds_rw.access_key[-5:]} ..{creds_rw.secret_key[-5:]}")
+
+    queue = get_queue(queue)
+
+    successes = 0
+    errors = 0
+
+    client = None
+
+    if verbose:
+        print("Starting local Dask cluster")
+
+    client = start_local_dask(threads_per_worker=threads, mem_safety_margin="1G")
+
+    for message in get_messages(queue, limit):
+        try:
+            task_def = get_task_from_message(message)
+            _task = rdr.stream([task_def], product)
+            configure_s3_access(aws_unsigned=True, cloud_defaults=True, client=client)
+            if verbose:
+                print(client)
+
+            results = process_tasks(
```

Comment:
so this is always one task in this list?

Reply:
Yes, that's right.  The name is plural for compatibility with your existing tools (otherwise this has to change in proc.py and will break your commands).